### PR TITLE
RequestLimitExceeded is now treated as ERR_INFRA_RATE_LIMITS_EXCEEDED

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area ops-productivity
/kind enhancement

**What this PR does / why we need it**:
`RequestLimitExceeded` is now treated as `ERR_INFRA_RATE_LIMITS_EXCEEDED`.

**Release note**:

```other operator
Error messages containing `RequestLimitExceeded` are now treated as `ERR_INFRA_RATE_LIMITS_EXCEEDED` (instead of `ERR_INFRA_QUOTA_EXCEEDED` before).
```

### CODE REVIEW

File: pkg/apis/core/v1beta1/helper/errors.go [hunk 1]
```diff
@@ -54,8 +54,8 @@ func (e *ErrorWithCodes) Error() string {
 
 var (
 	unauthorizedRegexp                  = regexp.MustCompile(`(?i)(Unauthorized|InvalidClientTokenId|InvalidAuthenticationTokenTenant|SignatureDoesNotMatch|Authentication failed|AuthFailure|AuthorizationFailed|invalid character|invalid_grant|invalid_client|Authorization Profile was not found|cannot fetch token|no active subscriptions|InvalidAccessKeyId|InvalidSecretAccessKey|query returned no results|UnauthorizedOperation|not authorized|InvalidSubscriptionId)`)
-	quotaExceededRegexp                 = regexp.MustCompile(`(?i)(LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met)`)
-	rateLimitsExceededRegexp            = regexp.MustCompile(`(?i)(Throttling|Too many requests)`)
+	quotaExceededRegexp                 = regexp.MustCompile(`(?i)([^Request]LimitExceeded|Quotas|Quota exceeded|exceeded quota|Quota has been met)`)
+	rateLimitsExceededRegexp            = regexp.MustCompile(`(?i)(RequestLimitExceeded|Throttling|Too many requests)`)
```
Comment (vpnachev) [inline pkg/apis/core/v1beta1/helper/errors.go]: Can you add a test about this?

Comment (rfranzke) [reply pkg/apis/core/v1beta1/helper/errors.go]: We only have basic tests for these error mappings and decided not to pollute them with the many hundred combinations of potential error messages that are expected to be mapped correctly. Hence, yes, I could do it, but I don't see why this should be special now, so I'd rather refrain from doing this (unless we do it consistently for all the error expressions).

Comment (vpnachev) [reply pkg/apis/core/v1beta1/helper/errors.go]: I think one line test here https://github.com/gardener/gardener/blob/b43d545d1957318dee1a17116707251ca7fef405/pkg/apis/core/v1beta1/helper/errors_test.go#L38-L63 will be useful to verify that `RequestLimitExceeded` error is not detected as quota exceeded error.

**Review by vpnachev — APPROVED** [review]
/lgtm

**Review by timebertt — APPROVED** [review]
/lgtm

**Pull request merged**